[Bug] `make px4_fmu-v6c_neural` failed

### Describe the bug

Bug: Enabling `CONFIG_LIB_TFLM` on NuttX fails to build because libstdc++ headers conflict with PX4’s NuttX header configuration. With the stock `px4_impl_os.cmake`,` -nostdinc++ plus _SYS_CDEFS_H_/_SYS_REENT_H_ `poison newlib headers, leading to `attempt to use poisoned "getenv" `when <cstdlib>/<stdlib.h> is pulled. If poison is disabled (`-DPX4_DISABLE_GCC_POISON`), compilation still fails because the same newlib <stdlib.h> can’t see the skipped definitions, producing `__uint32_t does not name a type`. The result is a hard stop during build of targets like `px4_fmu-v6c_neural`.



### To Reproduce

Build the NuttX target (`make px4_fmu-v6c_neural`) 
Optional mitigation tried: add` if(CONFIG_LIB_TFLM) add_definitions(-DPX4_DISABLE_GCC_POISON)` to px4_impl_os.cmake.

### Expected behavior

TFLM-enabled build should compile or at least not fail in standard library headers.

### Screenshot / Media

_No response_

### Flight Log

_No response_

### Software Version

_No response_

### Flight controller

_No response_

### Vehicle type

None

### How are the different components wired up (including port information)

_No response_

### Additional context

_No response_

@SindreMHegre 
Hi,

While building the TFLM module for the v6x board, the build failed in platforms/common/pab_manifest.c with:
    /px4/platforms/common/pab_manifest.c:480:22: error: implicit declaration of function ‘atoi’ [-Werror=implicit-function-declaration]
      int intval = atoi(val);

Adding `#include <stdlib.h>` in pab_manifest.c fixes the build. Could you confirm this is the correct fix and upstream it? Any concerns with this change?

Thanks!

--- a/docs/en/advanced/neural_networks.md
+++ b/docs/en/advanced/neural_networks.md
@@ -28,6 +28,10 @@ If you are looking for more resources to learn about the module, a website has b
 
 The module has been tested on a number of configurations, which can be build locally using the commands:
 
+::: warning
+This module requires Ubuntu 24 or newer, for older versions of Ubuntu please refer to this [guide](https://ntnu-arl.github.io/aerial_gym_simulator/9_sim2real/#installing-px4).
+:::
+
 ```sh
 make px4_sitl_neural
 ```

--- a/docs/en/advanced/neural_networks.md
+++ b/docs/en/advanced/neural_networks.md
@@ -29,7 +29,7 @@ If you are looking for more resources to learn about the module, a website has b
 The module has been tested on a number of configurations, which can be build locally using the commands:
 
 ::: warning
-This module requires Ubuntu 24 or newer, for older versions of Ubuntu please refer to this [guide](https://ntnu-arl.github.io/aerial_gym_simulator/9_sim2real/#installing-px4).
+This module requires Ubuntu 24.04 or newer (it is not supported in Ubuntu 22.04).
 :::
 
 ```sh

--- a/docs/en/advanced/neural_networks.md
+++ b/docs/en/advanced/neural_networks.md
@@ -26,11 +26,11 @@ If you are looking for more resources to learn about the module, a website has b
 
 ## Neural Network PX4 Firmware
 
-The module has been tested on a number of configurations, which can be build locally using the commands:
-
 ::: warning
 This module requires Ubuntu 24.04 or newer (it is not supported in Ubuntu 22.04).
 :::
+
+The module has been tested on a number of configurations, which can be build locally using the commands:
 
 ```sh
 make px4_sitl_neural

